### PR TITLE
fixed issue #41:

### PR DESCRIPTION
**Pull request by BorzdeG**
0) run initDb only if dataDir empty and not temporary directory
1) remove dataDir only if temporary dir

### CODE REVIEW

File: src/main/java/ru/yandex/qatools/embed/postgresql/PostgresProcess.java [hunk 1]
```diff
@@ -189,6 +189,12 @@ protected void onBeforeProcess(IRuntimeConfig runtimeConfig)
             throws IOException {
         super.onBeforeProcess(runtimeConfig);
         PostgresConfig config = getConfig();
+
+        final File     dbDir   = config.storage().dbDir();
+        if (dbDir.exists() && dbDir.listFiles() != null && dbDir.listFiles().length > 0) {
```
Comment (smecsia) [inline src/main/java/ru/yandex/qatools/embed/postgresql/PostgresProcess.java]: I'd prefer to have a conditional `runCmd` here rather than a conditional return. Conditional return would make more sense if we had more than 1 LOC here.

File: src/test/java/ru/yandex/qatools/embed/postgresql/TestPostgresStoredData.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import static java.lang.String.format;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static ru.yandex.qatools.embed.postgresql.EmbeddedPostgres.*;
+import static ru.yandex.qatools.embed.postgresql.util.SocketUtil.findFreePort;
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class TestPostgresStoredData {
```
Comment (smecsia) [inline src/test/java/ru/yandex/qatools/embed/postgresql/TestPostgresStoredData.java]: Let's add the ability to specify storage directory into `EmbeddedPostgres` and then simplify this test.

File: src/main/java/ru/yandex/qatools/embed/postgresql/PostgresProcess.java (hunk 1 above)
Comment (BorzdeG) [inline src/main/java/ru/yandex/qatools/embed/postgresql/PostgresProcess.java]: This is legal, because Initially de.flapdoodle.embed.process.runtime.AbstractProcess#onBeforeProcess() is completely empty

File: src/test/java/ru/yandex/qatools/embed/postgresql/TestPostgresStoredData.java (hunk 2 above)
Comment (BorzdeG) [inline src/test/java/ru/yandex/qatools/embed/postgresql/TestPostgresStoredData.java]: This is what I do in the setUpClass () method, providing an isolation from the environment in which the test is run

[russian]
именно это я и делаю в методе setUpClass(), обеспечивая отвязку от особенностей окружения, в котором запускается тест

Comment (smecsia) [inline src/test/java/ru/yandex/qatools/embed/postgresql/TestPostgresStoredData.java]: Yes, but it would much better to be able to bypass this directory to the `PostgresConfig`,  which is created [here](https://github.com/yandex-qatools/postgresql-embedded/blob/master/src/main/java/ru/yandex/qatools/embed/postgresql/EmbeddedPostgres.java#L124)

Comment (BorzdeG) [inline src/test/java/ru/yandex/qatools/embed/postgresql/TestPostgresStoredData.java]: Allow me to do it in a separate PR? :)
Since In the current PR error correction, and `new EmbeddedPostgres (version, dataDir);` this is an extension of the functionality

look at https://github.com/BorzdeG/postgresql-embedded/blob/2587d2c34ccce143c638292a01a415039c61da54/src/test/java/ru/yandex/qatools/embed/postgresql/TestPostgresStoredData.java#L71

Comment (smecsia) [inline src/test/java/ru/yandex/qatools/embed/postgresql/TestPostgresStoredData.java]: Deal. Let's do it in a separate PR.

Comment (BorzdeG) [inline src/test/java/ru/yandex/qatools/embed/postgresql/TestPostgresStoredData.java]: created PR #94
